include/emitc/emitc_tosa.h

```diff
@@ -98,6 +98,83 @@ inline Src mul(Src x, Src y, const int32_t shift) {
 }
 
 /// Other ops
+// Conv2dOp
+template <typename Dest, typename Src, typename Weights>
+Dest conv2d(Src input, Weights weights, Tensor1D<int64_t, 4> padding,
+            Tensor1D<int64_t, 2> stride, Tensor1D<int64_t, 2> dilation) {
+  static_assert(is_tensor_of_dim<4, Src>::value,
+                "Expected 4 dimensional input");
+  static_assert(is_tensor_of_dim<4, Dest>::value,
+                "Expected 4 dimensional output");
+  static_assert(is_tensor_of_dim<4, Weights>::value,
+                "Expected 4 dimensional weights");
```

Comment:
If we only support 4D Tensors here, why are we using a template?

Reply:
There is still `conv3d`. I hope that we can reuse this code/form a common base for both operations ;)

Reply:
In that case, please add a comment why this is a template + static asserts. I think without this information in mind it is a little confusing and simply looks wrong.

Reply:
We cannot use `Tensor4D` here because we don't know the size of the operands!

Reply:
@marbre Using these static_assert generates more clear error messages than the compiler does IMO. We use the same technique in the MHLO Convolution implementation as well.

Reply:
That comparison does not hold, as TOSA IR also restricts it to Tensor4D, whereas MHLO does not. The latter is the reason for inserting those into our MHLO header only implementation. But we're actually already discussing this *offline*, whatever this means in the current pandemic situation :laughing: